pkg/cli/admin/sosreport/sosreport.go

```diff
@@ -0,0 +1,687 @@
+	}
+}
+
+// newPod creates a pod with 2 containers with a shared volume mount:
+// - sosreport: init containers that run gather command
+// - copy: no-op container we can exec into
+func (o *SOSReportOptions) newPod(node, image string) *corev1.Pod {
+	zero := int64(0)
+
+	nodeSelector := map[string]string{
+		corev1.LabelOSStable: "linux",
+	}
+	if node == "" {
+		nodeSelector["node-role.kubernetes.io/master"] = ""
+	}
+	// Reused values
+	volumeName := "sosreport-output"
+	sosreportAdditionalArgs := ""
+	directoryType := corev1.HostPathDirectory
+	privileged := true
+	image = "registry.redhat.io/rhel8/support-tools:8.4-10"
+	ret := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "sosreport-collect-",
+			Labels: map[string]string{
+				"app": "sosreport",
+			},
+		},
+		Spec: corev1.PodSpec{
+
+			NodeName: node,
+			// This pod is ok to be OOMKilled but not preempted. Following the conventions mentioned at:
+			// https://github.com/openshift/enhancements/blob/master/CONVENTIONS.md#priority-classes
+			// so setting priority class to system-cluster-critical
+			PriorityClassName: "system-cluster-critical",
+			RestartPolicy:     corev1.RestartPolicyNever,
+			Volumes: []corev1.Volume{
+				{
+					Name: volumeName,
+					VolumeSource: corev1.VolumeSource{
+						EmptyDir: &corev1.EmptyDirVolumeSource{},
+					},
+				},
+				{
+					Name: "run",
+					VolumeSource: corev1.VolumeSource{
+						HostPath: &corev1.HostPathVolumeSource{
+							Path: "/run",
+							Type: &directoryType,
+						},
+					},
+				}, {
+					Name: "var-log",
+					VolumeSource: corev1.VolumeSource{
+						HostPath: &corev1.HostPathVolumeSource{
+							Path: "/var/log",
+							Type: &directoryType,
+						},
+					},
+				}, {
+					Name: "machine-id",
+					VolumeSource: corev1.VolumeSource{
+						HostPath: &corev1.HostPathVolumeSource{
+							Path: "/etc/machine-id",
+						},
+					},
+				}, {
+					Name: "local-time",
+					VolumeSource: corev1.VolumeSource{
+						HostPath: &corev1.HostPathVolumeSource{
+							Path: "/etc/localtime",
+						},
+					},
+				}, {
+					Name: "root-dir",
+					VolumeSource: corev1.VolumeSource{
+						HostPath: &corev1.HostPathVolumeSource{
+							Path: "/",
+							Type: &directoryType,
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name:            "sosreport",
+					Image:           image,
+					ImagePullPolicy: corev1.PullIfNotPresent,
+					// TODO: Add optional SOSReport flags here. (such as -k crio.all=on -k crio.logs=on)
+					// always force disk flush to ensure that all data gathered is accessible in the copy container
+					Command: []string{"/bin/bash", "-c", fmt.Sprintf("/usr/sbin/sos report --batch --tmp-dir=%s %s; sync", o.SourceDir, sosreportAdditionalArgs)},
+
+					// SOSReport requires additional securityContexts and mounts
+					SecurityContext: &corev1.SecurityContext{
+						Privileged: &privileged,
+					},
+					// Env: []corev1.EnvVar{
+					// 	{
```

Comment:
Currently the SOSReport is collected of the container itself. This is for quicker testing. To collect the SOSReport of the Node, uncomment these environment variables.